weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx

```diff
@@ -47,68 +75,238 @@ export const CallsPage: FC<{
   paginationModel: GridPaginationModel;
   setPaginationModel: (newModel: GridPaginationModel) => void;
 }> = props => {
+  const {entity, project, baseView, views, fetchViews} = props;
   const [filter, setFilter] = useControllableState(
     props.initialFilter ?? {},
     props.onFilterUpdate
   );
 
   const isEvaluationTable = useCurrentFilterIsEvaluationsFilter(
     filter,
-    props.entity,
-    props.project
+    entity,
+    project
   );
+  // table is the internal id stored in the object.
+  const table = isEvaluationTable ? 'evaluations' : 'traces';
+
+  const getTsClient = useGetTraceServerClientContext();
+
+  const tsClient = getTsClient();
+  const view = baseView.object_id;
 
-  const title = useMemo(() => {
-    if (isEvaluationTable) {
-      return 'Evaluations';
+  const currentViewDefinition = useCurrentViewDefinition(baseView);
+
+  const history = useHistory();
+  const onLoadView = (viewToLoad: TraceObjSchema) => {
+    // We want to preserve any params that are not part of view definition,
+    // e.g. peek drawer state.
+    const newQuery = new URLSearchParams(history.location.search);
+
+    // Clear out any params related to saved views
+    for (const key of SAVED_PARAM_KEYS) {
+      newQuery.delete(key);
     }
-    if (filter.opVersionRefs?.length === 1) {
-      const opName = opVersionRefOpName(filter.opVersionRefs[0]);
-      if (opName) {
-        return opNiceName(opName) + ' Traces';
-      }
+
+    // Update with params from the view definition
+    for (const [key, value] of Object.entries(viewToLoad.val.definition)) {
+      newQuery.set(key, JSON.stringify(value));
+    }
+
+    newQuery.set('view', viewToLoad.object_id);
+    history.push({search: newQuery.toString()});
+    props.onRecordLastView(viewToLoad.object_id);
```

Comment:
will this line run in an unmounted? I am not sure if the execution order ensures this runs before the URL changes